lsl_autobids/main.py

```diff
@@ -1,119 +1,130 @@
-            print('Exiting program')
-            exit() 
-        print(f'project/{project_name} exists')
+    if not project_path.exists():
+        raise FileNotFoundError(f"Project directory '{project_path}' does not exist.")
+    logger.info(f"Project folder for {project_name} found.")
+    
+    # update the project name in the project configuration file
+    update_project_config(project_path, project_name)
+
+    if cli_args.yes:
+        # Automatically proceed without asking
+        check_for_new_data()
+        print("Data check completed.")
+    else:
+        # Ask the user before proceeding
+        user_input = get_user_input("Do you want to check for new data in the project folder? (y/n): ")
+        if user_input == 'y':
+            check_for_new_data()
+            print("Data check completed.")
+        else:
+            print("Data check skipped by user.")
+            sys.exit(0)
+    # Ask the user if they want to check for new data
+    
 
-        user_input = get_user_input_for_data_check()
+def list_directories(path: str) -> list:
+    """List all directories at the given path.
 
-        if user_input=="y" or user_input=="Y":
-            check_for_new_data(project_name)
-        elif user_input == "n" or user_input == "N":
-            print("Program aborted. Closing...")
-            exit()
-    else: 
-        print('Project name not specified correctly')
-        exit()
-    return True
+    Args:
+        path (str): The path to scan.
 
+    Returns:
+        list: List of directory names.
 
-def list_directories(path):
+    Raises:
+        FileNotFoundError: If the path does not exist.
+        PermissionError: If access is denied.
     """
-    List all the directories in the path
-
-    Parameters
-    ----------
-    path : str
-        Path to the directory
-    Returns
-    -------
-    list
-        List of directories
+    try:
+        return [d for d in os.listdir(path) if os.path.isdir(os.path.join(path, d))]
+    except FileNotFoundError:
+        logger.error(f"The directory '{path}' was not found.")
+        raise
+    except PermissionError:
+        logger.error(f"Permission denied when accessing '{path}'.")
+        raise
+
+def update_project_config(project_path: str, project_name: str):
+    """Updates the project TOML configuration file.
+
+    Args:
+        project_path (Path): Path to the project directory.
+        project_name (str): Name of the project.
+
+    Raises:
+        FileNotFoundError: If the config file is missing.
     """
-    directories = [d for d in os.listdir(path) if os.path.isdir(os.path.join(path, d))]
-    return directories
+    toml_path = os.path.join(project_path, f"{project_name}_config.toml")
+    if not Path(toml_path).exists():
+        raise FileNotFoundError(f"Config file '{toml_path}' not found.")
 
+    config = read_toml_file(toml_path)
+    #config.setdefault('Dataset', {})['title'] = project_name 
+    config['Dataset']['title'] = project_name
```

Comment:
If the title is overwirtten and equal to the project folder anyway. Why do we keep it around in the toml?

Reply:
*bump*